src/odemis/odemisd/test/mdupdater_test.py

```diff
@@ -0,0 +1,86 @@
+"""
+Created on 22 Feb 2022
+
+@author: Éric Piel
+
+Copyright © 2022 Éric Piel, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms 
+of the GNU General Public License version 2 as published by the Free Software 
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR 
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with 
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import numpy
+from odemis import model
+from odemis.driver import static
+from odemis.model import Microscope
+from odemis.odemisd.mdupdater import MetadataUpdater
+from odemis.util import mock
+import unittest
+import unittest.mock
+
+
+class MDUpdaterTest(unittest.TestCase):
+
+    def test_lens_observer(self):
+        # Create microscope (.Alive)
+        mic = Microscope("Fake mic", "secom")
+
+        # Create a CCD (the role matters)
+        img = model.DataArray(numpy.empty((512, 768), dtype=numpy.uint16))
+        ccd = mock.FakeCCD(img)  # role is always "ccd"
+
+        # Create Lens affects CCD (the role matters)
+        lens = static.OpticalLens("Lens", "lens", mag=0.51, pole_pos=(458, 519), focus_dist=0.5e-3,
+                                  mirror_pos_top=[600.5, 0.2], mirror_pos_bottom=(-200, 0.3))
+        lens.affects.value = [ccd.name]
+        
+        # Mock model.getComponent()
+        comps = [ccd, lens]
+        def fake_get_component(name):
+            for c in comps:
+                if c.name == name:
+                    return c
+            raise LookupError(f"no component {name}") 
+
+        with unittest.mock.patch.object(model, "getComponent", fake_get_component):
+
+            # Create a MetadataUpdater
+            mdup = MetadataUpdater("MDUpdater", mic)
```

Comment:
mdup is not used as a variable right?

```suggestion
            MetadataUpdater("MDUpdater", mic)
```

Reply:
Actually, although we don't explicitly use it, we (theoretically) need to hold the instance, because it's working in the background (whenever a VA changes). So (theoretically), if we immediately discard the return value (ie, the instance), Python could delete the object. In practice, it's quite unlikely because the instance probably has some self references, and so it's need to go through the garbage collector, which, compared to the amount of code executed after, rarely runs. 

Anyway... I've found a even better way: I've added an explicitly call to `mdup.terminate()` at the end of the test. So we test more code, and it also makes this variable used.

Reply:
That's great Eric, and thanks for the explanation!